internal/querycoord/task.go

```diff
@@ -320,60 +318,29 @@ func (lct *loadCollectionTask) execute(ctx context.Context) error {
 	if err != nil {
-		log.Error("loadCollectionTask: showPartition failed", zap.Int64("collectionID", collectionID), zap.Error(err))
+		log.Error("loadCollectionTask: showPartition failed", zap.Int64("collectionID", collectionID), zap.Int64("msgID", lct.Base.MsgID), zap.Error(err))
 		lct.setResultInfo(err)
 		return err
 	}
 
 	if showPartitionResponse.Status.ErrorCode != commonpb.ErrorCode_Success {
 		err = errors.New(showPartitionResponse.Status.Reason)
-		log.Error("loadCollectionTask: showPartition failed", zap.Int64("collectionID", collectionID), zap.Error(err))
+		log.Error("loadCollectionTask: showPartition failed", zap.Int64("collectionID", collectionID), zap.Int64("msgID", lct.Base.MsgID), zap.Error(err))
 		lct.setResultInfo(err)
 		return err
 	}
-	log.Debug("loadCollectionTask: get collection's all partitionIDs", zap.Int64("collectionID", collectionID), zap.Int64s("partitionIDs", showPartitionResponse.PartitionIDs))
 
-	partitionIDs := showPartitionResponse.PartitionIDs
-	toLoadPartitionIDs := make([]UniqueID, 0)
-	hasCollection := lct.meta.hasCollection(collectionID)
-	watchPartition := false
-	if hasCollection {
-		watchPartition = true
-		loadType, _ := lct.meta.getLoadType(collectionID)
-		if loadType == querypb.LoadType_loadCollection {
-			for _, partitionID := range partitionIDs {
-				hasReleasePartition := lct.meta.hasReleasePartition(collectionID, partitionID)
-				if hasReleasePartition {
-					toLoadPartitionIDs = append(toLoadPartitionIDs, partitionID)
-				}
-			}
-		} else {
-			for _, partitionID := range partitionIDs {
-				hasPartition := lct.meta.hasPartition(collectionID, partitionID)
-				if !hasPartition {
-					toLoadPartitionIDs = append(toLoadPartitionIDs, partitionID)
-				}
-			}
-		}
-	} else {
-		toLoadPartitionIDs = partitionIDs
-	}
-
-	log.Debug("loadCollectionTask: toLoadPartitionIDs", zap.Int64s("partitionIDs", toLoadPartitionIDs))
-	lct.meta.addCollection(collectionID, lct.Schema)
-	lct.meta.setLoadType(collectionID, querypb.LoadType_loadCollection)
-	for _, id := range toLoadPartitionIDs {
-		lct.meta.addPartition(collectionID, id)
-	}
+	toLoadPartitionIDs := showPartitionResponse.PartitionIDs
+	log.Debug("loadCollectionTask: get collection's all partitionIDs", zap.Int64("collectionID", collectionID), zap.Int64s("partitionIDs", toLoadPartitionIDs), zap.Int64("msgID", lct.Base.MsgID))
 
 	loadSegmentReqs := make([]*querypb.LoadSegmentsRequest, 0)
 	watchDmChannelReqs := make([]*querypb.WatchDmChannelsRequest, 0)
-	channelsToWatch := make([]string, 0)
-	var watchDeltaChannels []*datapb.VchannelInfo
+	var deltaChannelInfos []*datapb.VchannelInfo
+	var dmChannelInfos []*datapb.VchannelInfo
 	for _, partitionID := range toLoadPartitionIDs {
-		dmChannelInfos, binlogs, err := getRecoveryInfo(lct.ctx, lct.dataCoord, collectionID, partitionID)
+		vChannelInfos, binlogs, err := getRecoveryInfo(lct.ctx, lct.dataCoord, collectionID, partitionID)
```

Comment:
Probably  need to batch getRecoveryInfo if partition id is a long list, so far I'm ok with this